File: project-signer/src/test/java/org/jesperancinha/parser/projectsigner/service/OptionsServiceMock.java
package org.jesperancinha.parser.projectsigner.service;

import org.jesperancinha.parser.markdowner.filter.ReadmeNamingParser;
import org.jesperancinha.parser.markdowner.filter.ReadmeNamingParser.ReadmeNamingParserBuilder;
import org.jesperancinha.parser.projectsigner.configuration.ProjectSignerOptions;
import org.mockito.Mockito;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

import java.nio.file.Path;
import java.util.Objects;

import static org.jesperancinha.parser.projectsigner.configuration.ProjectSignerOptionsTest.README_MD;
import static org.jesperancinha.parser.projectsigner.configuration.ProjectSignerOptionsTest.TEMPLATE_LOCATION_README_MD;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@Service
@Profile({"test", "localtest", "default"})
public class OptionsServiceMock {
    private ProjectSignerOptions projectSignerOptions;
    private ReadmeNamingParserBuilder commonBuilder;

    public void setNoEmptyUp() {
        Mockito.clearInvocations(projectSignerOptions);
        when(projectSignerOptions.isNoEmpty()).thenReturn(true);
        commonBuilder = ReadmeNamingParser.builder()
                .templateLocation(this.projectSignerOptions.getTemplateLocation())
                .isNoEmpty(this.projectSignerOptions.isNoEmpty());
    }

    public ProjectSignerOptions processOptions() {
        final String rootPath = Objects.requireNonNull(getClass().getResource("/dummyDirectory")).getPath();
        final ProjectSignerOptions projectSignerOptions = mock(ProjectSignerOptions.class);
        when(projectSignerOptions.getRootDirectory()).thenReturn(Path.of(rootPath));
        when(projectSignerOptions.getTemplateLocation()).thenReturn(Path.of(README_MD));
        when(projectSignerOptions.getTagNames()).thenReturn(new String[]{"License", "About me"});
        this.projectSignerOptions = projectSignerOptions;
        this.commonBuilder = ReadmeNamingParser.builder()
                .templateLocation(this.projectSignerOptions.getTemplateLocation())
                .isNoEmpty(this.projectSignerOptions.isNoEmpty());
        return projectSignerOptions;
    }

    public ProjectSignerOptions getProjectSignerOptions() {
        return projectSignerOptions;
    }

    public ReadmeNamingParserBuilder getCommonNamingParser() {
        return commonBuilder;
    }

    public void setNoEmptyDown() {
        Mockito.clearInvocations(projectSignerOptions);
        when(projectSignerOptions.isNoEmpty()).thenReturn(false);
        commonBuilder = ReadmeNamingParser.builder()
                .templateLocation(this.projectSignerOptions.getTemplateLocation())
                .isNoEmpty(this.projectSignerOptions.isNoEmpty());
    }
}

File: project-signer-runnable/src/test/kotlin/org/jesperancinha/parser/projectsigner/service/OptionsServiceMock.kt
package org.jesperancinha.parser.projectsigner.service

import org.jesperancinha.parser.markdowner.filter.ReadmeNamingParser
import org.jesperancinha.parser.projectsigner.configuration.ProjectSignerOptions
import org.jesperancinha.parser.projectsigner.configuration.ProjectSignerOptionsTest
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service
import java.nio.file.Path
import java.util.*

@Service
@Profile("test", "localtest", "default")
open class OptionsServiceMock (
    var projectSignerOptions: ProjectSignerOptions?= null,
    private var commonBuilder: ReadmeNamingParser?= null
) {

    fun setNoEmptyUp() {
        projectSignerOptions?.noEmpty = true
        commonBuilder = projectSignerOptions?.templateLocation?.let {
            ReadmeNamingParser(
                templateLocation = it,
                isNoEmpty = (projectSignerOptions?.noEmpty == true)
            )
        }

    }

    fun processOptions(): ProjectSignerOptions {
        val rootPath = Objects.requireNonNull(javaClass.getResource("/dummyDirectory")).path
        val projectSignerOptions = ProjectSignerOptions()
        projectSignerOptions.rootDirectory= Path.of(rootPath)
        projectSignerOptions.templateLocation=Path.of(ProjectSignerOptionsTest.README_MD)
        projectSignerOptions.tagNames=arrayOf("License", "About me")
        this.projectSignerOptions = projectSignerOptions
        commonBuilder = this.projectSignerOptions?.noEmpty?.let {
            this.projectSignerOptions!!.templateLocation?.let { it1 ->
                ReadmeNamingParser(
                    templateLocation = it1,
                    isNoEmpty = it
                )
            }
        }
        return projectSignerOptions
    }

    fun setNoEmptyDown() {
        commonBuilder = projectSignerOptions?.noEmpty?.let {
            projectSignerOptions?.templateLocation?.let { it1 ->
                ReadmeNamingParser(
                    templateLocation = it1,
                    isNoEmpty = it
                )
            }
        }
    }
}